### PR TITLE
fix: Move progressbar role to spinner root slot and update tabindex documentation

### LINKED ISSUE #22662: [Bug]: v9 Spinner: Primary with label: Elements must only use allowed ARIA attributes
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
vr-tests storybook
```

### Are you reporting Accessibility issue?

yes

### Reproduction

vr-tests storybook

## Actual Behavior
Primary with label: Elements must only use allowed ARIA attributes (ARIA attribute cannot be used, add a role attribute or use a different element: aria-labelledby) 

## Expected Behavior
he label needs to move to the element with role progressbar

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR is to address some accessibility bugs in Spinner. It does two things:

- Moves the `role=progressbar` attribute from the spinner slot to the root slot.
- Updates `tabindex` information to be more accurate for screen readers

Fixes #22662 and #23353

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                             |   Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                               Change |
| :-------------------------------------------------------------------------------------------- | -------------------------: | --------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr> | `18.15 kB`<br />`5.989 kB` | `18.159 kB`<br />`5.992 kB` | `9 B` <br />`3 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `183.617 kB`<br />`50.924 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.988 kB`<br />`11.108 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6b87db7b177abe1ae8e1f9bd1938b4c29e86acb6'>6b87db7b177abe1ae8e1f9bd1938b4c29e86acb6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/Avatar-mount_base.html">885</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/Avatar-mount.html">915</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/Button-mount_base.html">570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/Button-mount.html">571</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProvider-mount_base.html">1872</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProvider-mount.html">1824</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-mount_base.html">283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-mount.html">271</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">253</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">228</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">287</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">325</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/MakeStyles-mount_base.html">1581</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23385/perf-test-react-components/MakeStyles-mount.html">1582</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-spinner/src/components/Spinner/Spinner.test.tsx [hunk 1]
```diff
@@ -32,11 +32,11 @@ describe('Spinner', () => {
 
   it('doesnt render Spinner when slot is overridden', () => {
```
Comment (khmakoto) [inline packages/react-components/react-spinner/src/components/Spinner/Spinner.test.tsx:33]: I think the name of these tests should probably be changed.

Comment (smhigley) [reply packages/react-components/react-spinner/src/components/Spinner/Spinner.test.tsx:33]: I think it'd make sense to keep the name and change the test to query for the SVG and verify it's not there

**Review by smhigley — APPROVED** [review]
Aside from Mak's comment, it looks good to me!

**Review by smhigley — CHANGES_REQUESTED** [review]
Ah, oops, forgot one thing -- `aria-labelledby` should also be moved to the root slot, it's still applied to the spinner slot by default now.

**Review by smhigley — APPROVED** [review]

**Pull request merged**